### PR TITLE
Extend the sidebar to see keywords by either vocabulary or alphabetically

### PR DESCRIPTION
**Pull request by Era-cell**
@jviotti I have created the new branch for grouping alphabetical ordering to avoid conflicts. You can check on this once.
Re-opening this https://github.com/Intelligence-AI/learnjsonschema.com/pull/233 PR

### CODE REVIEW

File: assets/main.js [hunk 1]
```diff
@@ -31,3 +31,50 @@ function copyToClipboard(text, button) {
...
+  localStorage.setItem('sidebarGrouping', 'vocabulary');
+  setSidebarGrouping('vocabulary');
+});
+
+alphabeticTab.addEventListener('click', function () {
+  localStorage.setItem('sidebarGrouping', 'alphabetic');
+  setSidebarGrouping('alphabetic');
+});
+
+document.addEventListener("DOMContentLoaded", function () {
```
Comment (jviotti) [inline assets/main.js:71]: Isn't this duplicated?

**Comment by jviotti** [comment]
@Era-cell The duplicates are gone, but now they are not showing up side by side anymore, but one above the other. Maybe you accidentally changed something else?

Screenshot 2024-06-20 at 9 31 02 PM

**Comment by jviotti** [comment]
Also, I think there is something off with the alphabetic view. Here I'm focusing on 2020-12, but I'm still seeing 2019-09 keywords (like `$recursiveRef`)

Screenshot 2024-06-20 at 9 32 21 PM

File: assets/main.js (hunk 1 above)
Comment (Era-cell) [reply assets/main.js:71]: Yeah, my bad

**Comment by Era-cell** [comment]
> Also, I think there is something off with the alphabetic view. Here I'm focusing on 2020-12, but I'm still seeing 2019-09 keywords (like `$recursiveRef`)
> 
> Screenshot 2024-06-20 at 9 32 21 PM
> Even when breaking up by vocabulary:
> 
> Screenshot 2024-06-20 at 9 33 02 PM

I don't know how the line to filter the dialect got missed, I have added it again

**Comment by Era-cell** [comment]
> @Era-cell The duplicates are gone, but now they are not showing up side by side anymore, but one above the other. Maybe you accidentally changed something else?
> 
> Screenshot 2024-06-20 at 9 31 02 PM

This is caused when styling is removed for this. I think re-rendering scss will solve it...

**Comment by jviotti** [comment]
@Era-cell The keywords were fixed, but the nav still doesn't show side by side for me. Let me dig into it a bit

**Comment by jviotti** [comment]
It's the font size, at least for my browser. I'll quickly fix it myself by adding a new commit to this PR so we can merge

**Pull request merged**